mockgen/internal/tests/mock_in_test_package/README.md

```diff
@@ -0,0 +1,18 @@
+When a mock is put into the test package (e.g., `foo` vs `foo_test`), any type
```

Comment:
This description reads well right now that the bug exists but in the future it will be confusing to understand. It would be better to write it from that perspective.

It should simply explain the situation it is testing. You can mention that this used to be broken but that will be of little interest the moment this is committed

Reply:
How about something far simpler:

> Test the case where the package has the `_test` suffix.